src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,31 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+
+type Props = {
+  selectedTodo: Todo;
+  handleClose?: (todo: Todo | null) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo: { title, id, user, completed },
+  handleClose: onClose = () => {},
+}) => {
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getUser(user?.id || 0)
+      .then(() => setIsLoading(false))
+      .catch(() => setIsLoading(false));
+  }, [user?.id]);
```

Comment:
You're directly interacting with the DOM by manipulating the loading state. React promotes a declarative style of programming where you describe what you want to happen, not how it should happen. It would be better to handle loading state declaratively using React state.

Reply:
i'm literally using states in this code, where do i manipulating DOM?